src/components/VHeader/VFilterButton.vue

```diff
@@ -0,0 +1,124 @@
+    <span class="filter-label">{{ buttonLabel }}</span>
+  </VButton>
+</template>
+
+<script>
+import FilterIcon from '~/assets/icons/filter.svg?inline'
+import { computed, defineComponent, useContext } from '@nuxtjs/composition-api'
+import { useMediaQuery } from '~/composables/use-media-query'
+import { SEARCH } from '~/constants/store-modules'
+import { SET_FILTER_IS_VISIBLE } from '~/constants/mutation-types'
+
+const VFilterButton = defineComponent({
+  name: 'FilterButton',
+  components: {
+    FilterIcon,
+  },
+  props: {
+    /**
+     * Whether the page has been scrolled. If true on mobile, the button
+     * label is blank.
+     * This is currently not implemented, and is always false.
+     * TODO(@obulat): add use-scroll composable to VHeader
+     * @default false
+     */
+    isScrolled: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  setup(props) {
+    const { i18n, store } = useContext()
+    const { isScrolled = false } = props
+
+    const defaultWindow = typeof window !== 'undefined' ? window : undefined
+    const isTablet = useMediaQuery('(min-width: 640px)', defaultWindow)
+
+    const appliedFilterTags = computed(
+      () => store.getters['search/appliedFilterTags']
+    )
+    const isAnyFilterApplied = computed(
+      () => appliedFilterTags.value.length > 0
+    )
+
+    /**
+     * On screens wider than tablet: 'Filters'.
+     * On mobile screens:
+     * 1. If no filters are applied - '' (icon only).
+     * 2. If the page is scrolled down - '' (icon only).
+     * 3. If the page wasn't scrolled, and some filters are applied:
+     * the number of applied filters, e.g. '7'.
+     *
+     * @type {ComputedRef<string>}
+     */
+    const buttonLabel = computed(() => {
+      if (isTablet.value) {
+        return i18n.t('header.filter-button.simple')
+      }
+      return isScrolled.value || !isAnyFilterApplied.value
+        ? ''
+        : i18n.tc(
+            'header.filter-button.with-count',
+            appliedFilterTags.value.length
+          )
+    })
+
+    /**
+     * Icon is not shown on mobile if no filters were applied on mobile.
+     *
+     * @type {ComputedRef<boolean>}
+     */
+    const showIcon = computed(() => {
+      if (isTablet.value) {
+        return true
+      }
+      return !isAnyFilterApplied.value
+    })
+
+    /**
+     * The sidebar visibility state is kept in the Vuex store. The button is
+     * set to `pressed` state when the sidebar is visible.
+     *
+     * @type {ComputedRef<boolean>}
+     */
+    const sidebarVisible = computed(() => {
+      return store.state.search.isFilterVisible
+    })
+    const toggleFilters = () => {
+      store.commit(`${SEARCH}/${SET_FILTER_IS_VISIBLE}`, {
+        isFilterVisible: !store.state.search.isFilterVisible,
+      })
+    }
+
+    return {
+      buttonLabel,
+      sidebarVisible,
+      showIcon,
+
+      toggleFilters,
+    }
+  },
+})
+
+export default VFilterButton
+</script>
+
+<style>
```

Comment:
Can we use `scoped` or `module` here?

Reply:
The style was blank, so I removed it completely.